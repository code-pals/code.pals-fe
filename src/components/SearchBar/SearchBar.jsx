import {
  Input,
  Box,
  Stack,
  InputGroup,
  InputLeftElement,
  Icon,
  Center,
} from '@chakra-ui/react';
import { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { searchPostsAndComments } from '../../services/fetch-utils';
import CommentBox from '../CommentBox/CommentBox';
import PostHomeBox from '../PostHomeBox/PostHomeBox';

export default function SearchBar() {
  const [search, setSearch] = useState('');
  const [comments, setComments] = useState([]);
  const [posts, setPosts] = useState([]);
  const history = useHistory();

  const handleSubmit = async (e) => {
    e.preventDefault();
    const responseArr = await searchPostsAndComments(search);
    console.log(responseArr);
    setComments(responseArr.body[0]);
    setPosts(responseArr.body[2]);
    // history.push({
    //   pathname: '/results',
    //   state: comments })
  };

  console.log(comments, posts);
  return (
    <>
      <Center>
        <Box m="20px" width="50%">
          <Stack spacing={4}>
            <form onSubmit={handleSubmit}>
              <InputGroup>
                <InputLeftElement
                  children={<Icon name="search" color="gray.300" />}
                />
                <Input
                  type="text"
                  placeholder="Search"
                  value={search}
                  onChange={(e) => setSearch(e.target.value)}
                />
              </InputGroup>
              <button type="submit">Submit</button>
            </form>
          </Stack>
        </Box>
      </Center>
      <div>
        {posts.map((post) => {
          return <PostHomeBox key={post.postId} post={post} />;
        })}

        {comments.map((comment) => {
          return <CommentBox key={comment.commentId} comment={comment} />;
        })}
      </div>
    </>
  );
}
