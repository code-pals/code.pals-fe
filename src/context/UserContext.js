import { createContext, useContext, useEffect, useMemo, useState } from 'react';
import { fetchUser } from '../services/users';

const request = require('superagent');

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [loading, setLoading] = useState(true);
  const [user, setUser] = useState({});

 /* useEffect(() => {
    (async () => {
      const fetchedUser = await fetchUser();
      console.log(fetchedUser);
      setUser(fetchedUser);
      setLoading(false);
    })();
  }, []);
*/
  async function logIn() {
    window.location.assign(`${process.env.URL}/api/v1/users/login`);
  }

  async function logOut() {
    const res = await request
      .delete(`${process.env.URL}/api/v1/users/sessions`)
      .withCredentials();

    localStorage.removeItem('storageUser'); 
    setUser({})
    ;
    history.push('/');
  }

  const value = useMemo(
    () => ({ user, setUser, logIn, logOut, loading }),
    [user]
  );

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }
  return context;
};

export { UserContext, UserProvider, useUser };
