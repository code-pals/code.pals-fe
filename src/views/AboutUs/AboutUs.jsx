import ProfileBox from '../../components/ProfileBox/ProfileBox.jsx';
import about from '../../components/ProfileBox/about.json';

export default function AboutUs() {
  console.log(about);
  return (
    <>
      <h1>About Us</h1>
      <ul>
        {about.map((profile, i) => {
          return (
            <li key={i}>
              {' '}
              <ProfileBox {...profile} />{' '}
            </li>
          );
        })}
      </ul>
    </>
  );
}
