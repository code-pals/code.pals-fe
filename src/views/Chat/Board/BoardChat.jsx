import React from 'react';
import Chat from '../Chat/Chat';
import { useParams } from 'react-router-dom';

export default function BoardChat({ socket }) {
  const { username } = useParams();
  const { roomname } = useParams();

  return (
    <>
      <div className="right">
        <Chat username={username} roomname={roomname} socket={socket} />
      </div>
    </>
  );
}
