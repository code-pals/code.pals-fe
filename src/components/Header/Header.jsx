import { ReactNode } from 'react';
import {
  Box,
  Flex,
  Avatar,
  HStack,
  IconButton,
  Button,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
  MenuDivider,
  useDisclosure,
  useColorModeValue,
  useColorMode,
  Stack,
} from '@chakra-ui/react';
import { MoonIcon, SunIcon } from '@chakra-ui/icons';
import { HamburgerIcon, CloseIcon, AddIcon } from '@chakra-ui/icons';
import { Link } from 'react-router-dom';
import { useUser } from '../../context/UserContext.js';

const Links = ['Home', 'Create', 'Messages', 'Login', 'Search', 'About Us'];

const NavLink = ({ children }) => (
  <Link
    px={2}
    py={1}
    rounded={'md'}
    _hover={{
      textDecoration: 'none',
      bg: useColorModeValue('gray.200', 'gray.700'),
    }}
    to={'#'}
  >
    {children}
  </Link>
);

export default function withAction() {
  const { colorMode, toggleColorMode } = useColorMode();
  const { isOpen, onOpen, onClose } = useDisclosure();
  const { user, getUser, logIn, logOut } = useUser();
  //const storedU = localStorage.getItem('storageUser');
  //const storedUser = JSON.parse(storedU);
  //console.log('headerStoredUser', storedUser);
  console.log('USER3', user);

  return (
    <>
      <Box bg={useColorModeValue('gray.100', 'gray.900')} px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <IconButton
            size={'md'}
            icon={isOpen ? <CloseIcon /> : <HamburgerIcon />}
            aria-label={'Open Menu'}
            display={{ md: 'none' }}
            onClick={isOpen ? onClose : onOpen}
          />
          <HStack spacing={8} alignItems={'center'}>
            <Box>Code.Pals</Box>
            <HStack
              as={'nav'}
              spacing={4}
              display={{ base: 'none', md: 'flex' }}
            >
              <Link to={'/'} underline="none">
                Home
              </Link>
              <Link to={'/create'} underline="none">
                Create
              </Link>
              {!user.github ? (
                <Link to={'/login'} underline="none">
                  Login
                </Link>
              ) : (
                <Link to={'/'} underline="none" onClick={logOut}>
                  Logout
                </Link>
              )}
              <Link to={'/results'} underline="none">
                Search
              </Link>
              <Link to={'/chatrooms'} underline="none">
                Chat
              </Link>
              <Link to={'/aboutus'} underline="none">
                About Us
              </Link>
             
              {user.github}
            </HStack>
          </HStack>
          <Flex alignItems={'center'}>
            {/* <Button
              variant={'solid'}
              colorScheme={'teal'}
              size={'sm'}
              mr={4}
              leftIcon={<AddIcon />}
            >
              Action
            </Button> */}
            <Button onClick={toggleColorMode} mr="10px">
              {colorMode === 'light' ? <MoonIcon /> : <SunIcon />}
            </Button>
            <Menu>
              <MenuButton
                as={Button}
                rounded={'full'}
                variant={'link'}
                cursor={'pointer'}
                minW={0}
              >
                {user.avatar ? <Avatar size={'sm'} src={user.avatar} /> : ''}
              </MenuButton>
              <MenuList>
                <Link to={`/profile/${user.github}`} underline="none">
                  {user.github}'s
                  <br />
                  Profile
                </Link>
                {/* <MenuItem>Link 2</MenuItem>
                <MenuDivider />
                <MenuItem>Link 3</MenuItem> */}
              </MenuList>
            </Menu>
          </Flex>
        </Flex>
        

        {isOpen ? (
          <Box pb={4} display={{ md: 'none' }}>
            <Stack as={'nav'} spacing={4}>
              <Link to={'/'} underline="none">
                Home
              </Link>
              <Link to={'/create'} underline="none">
                Create
              </Link>
              {!user.github ? (
                <Link to={'/login'} underline="none">
                  Login
                </Link>
              ) : (
                <Link to={'/'} underline="none" onClick={logOut}>
                  Logout
                </Link>
              )}
              <Link to={'/results'} underline="none">
                Search
              </Link>
              <Link to={'/chatrooms'} underline="none"  >
                Chat
              </Link>
              <Link to={'/aboutus'} underline="none">
                About Us
              </Link>
              {/* {Links.map((link) => (
                <NavLink key={link}>{link}</NavLink>
              ))} */}
            </Stack>
          </Box>
        ) : null}
      </Box>
    </>
  );
}
