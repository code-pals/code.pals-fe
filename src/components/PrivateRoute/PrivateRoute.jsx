import { Redirect, Route } from 'react-router-dom';
import { useUser } from '../../context/UserContext';

export default function PrivateRoute({ children, ...rest }) {
  const { user } = useUser();
  console.log('PRIVATEROUTEPAGE', user.github);
  return (
    <Route
      {...rest}
      render={({ location }) =>
        user.github ? (
          children
        ) : (
          <Redirect
            to={{
              pathname: '/login',
              state: { from: location },
            }}
          />
        )
      }
    />
  );
}
