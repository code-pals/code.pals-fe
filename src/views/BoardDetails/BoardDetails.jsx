import { Box, Button, Input, Text, Center } from '@chakra-ui/react';
import React, { useEffect, useState } from 'react';
import { Link, useParams } from 'react-router-dom';
import { deleteBoard, getBoardById } from '../../services/fetch-utils.js';
import BoardHomeBox from '../../components/BoardHomeBox/BoardHomeBox.jsx';
import { useUser } from '../../context/UserContext.js';
import { useHistory } from 'react-router-dom';
import BoardForm from '../../components/BoardForm/BoardForm.jsx';

export default function BoardDetails() {
  const { id } = useParams();
  const { user } = useUser();
  const history = useHistory();

  const [board, setBoard] = useState('');
  const [loading, setLoading] = useState(true);
  const [showForm, setShowForm] = useState(false);

  useEffect(() => {
    const singleBoard = async () => {
      const returnBoard = await getBoardById(id);
      setBoard(returnBoard.body);
      setLoading(false);
    };
    singleBoard();
  }, [id]);

  const handleDelete = async () => {
    const answer = confirm('Are you sure you want to delete this Board?');
    if (answer) {
      const response = await deleteBoard(id);
      history.push('/');
    }
  };
  const handleEdit = async () => {
    setShowForm((prev) => !prev);
  };

  return (
    <>
      <BoardHomeBox board={board} goals={board.goal} />
      <Button onClick={handleDelete}>Delete Board</Button>
      <Button onClick={handleEdit}>Edit Board</Button>
      {showForm && <BoardForm />}
      <Link target={'_blank'} to={`/chat/${board.title}/${user.github}`}>
        <Center>
          <Button>Enter Boardroom!</Button>
        </Center>
      </Link>
      {/* {board.title}
        {board.summary}
        {board.goal}
        {board.group_size} */}
    </>
  );
}
