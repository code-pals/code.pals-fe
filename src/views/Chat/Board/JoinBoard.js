import { Button, Heading, Input } from '@chakra-ui/react';
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
import { useUser } from '../../../context/UserContext';
// import './home.scss';

export default function JoinBoard({ socket }) {
  const { user } = useUser();
  const [roomname, setRoomname] = useState('');
  const history = useHistory();

  const sendData = () => {
    if (user.userId !== '' && roomname !== '') {
      history.push(`/chat/${roomname}/${user.github}`);
    } else {
      alert('username and roomname are must!!');
      window.location.reload();
    }
  };

  return (
    <>
      <Center>
        <Heading>Welcome to the chat, {user.github}</Heading>
        <h2></h2>
        <Input
          placeholder="Input Room"
          value={roomname}
          onChange={(e) => setRoomname(e.target.value)}
        />

        <Button onClick={sendData}>Join.</Button>
      </Center>
    </>
  );
}
