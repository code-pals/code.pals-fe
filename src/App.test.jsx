import { render } from '@testing-library/react';
import { UserProvider } from './context/UserContext.js';
import { ChakraProvider } from '@chakra-ui/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import Homepage from './views/Homepage/Homepage.jsx';
import Login from './views/Login/Login.jsx';
import Profile from './views/Profile/Profile.jsx';
import Header from './components/Header/Header.jsx';
import Create from './views/Create/Create.jsx';
import SearchBar from './components/SearchBar/SearchBar.jsx';
import AboutUs from './views/AboutUs/AboutUs.jsx';
import CreateProfile from './views/Profile/CreateProfile.jsx';
import PostDetails from './views/PostDetails/PostDetails.jsx';
import JoinBoard from './views/Chat/Board/JoinBoard.js';
import BoardChat from './views/Chat/Board/BoardChat.jsx';
import BoardDetails from './views/BoardDetails/BoardDetails.jsx';
import ChatRooms from './views/ChatRooms/ChatRooms.jsx';
import PrivateRoute from './components/PrivateRoute/PrivateRoute.jsx';
import OauthReturn from './views/OauthReturn/OauthReturn.jsx';
import socketIOClient from 'socket.io-client';
import MockedSocket from 'socket.io-mock';

jest.mock('socket.io-client');
jest.mock('../src/context/UserContext.js');

describe('it tests the entire app', () => {
  let socket;

  beforeEach(() => {
    socket = new MockedSocket();
    socketIOClient.mockReturnValue(socket);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders the homepage', () => {
    const mockUser = {
      user_id: 1,
      github: 'h',
      email: 's@s.com',
      username: 'alchemy',
      pronoun: 'they',
      experience: 'hard',
      tech: 'JS',
      avatar: 'https://placekitten.com/200/300',
      repos: 123,
      bio: 'stuff',
      member_since: '2020',
      created: Date.now(),
    };
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <ChakraProvider>
          <UserProvider mockUser={{ mockUser }}>
            <Header />
            <Switch>
              <Route exact path="/">
                <Homepage />
              </Route>
              <Route exact path="/login">
                <Login />
              </Route>
              <Route exact path="/profile/:username">
                <Profile />
              </Route>
              <PrivateRoute exact path="/create">
                <Create />
              </PrivateRoute>
              <Route exact path="/results">
                <SearchBar />
              </Route>
              <Route exact path="/aboutus">
                <AboutUs />
              </Route>
              <Route exact path="/oauthreturn">
                <OauthReturn />
              </Route>
              <Route exact path="/createprofile">
                <CreateProfile />
              </Route>
              <Route exact path="/postdetails/:id">
                <PostDetails />
              </Route>
              <Route exact path="/chat">
                <JoinBoard socket={socket} />
              </Route>
              <Route exact path="/chat/:roomname/:username">
                <BoardChat socket={socket} />
              </Route>
              <Route exact path="/boarddetails/:id">
                <BoardDetails />
              </Route>
              <Route exact path="/chatrooms">
                <ChatRooms />
              </Route>
            </Switch>
          </UserProvider>
        </ChakraProvider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
