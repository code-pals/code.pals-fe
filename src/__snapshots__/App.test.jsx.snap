// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the entire app renders the homepage 1`] = `
<div>
  <div
    class="css-1mr4ysp"
  >
    <div
      class="css-lygqz2"
    >
      <button
        aria-label="Open Menu"
        class="chakra-button css-v4wdal"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="chakra-stack css-1ocidfa"
      >
        <div
          class="css-0"
        >
          Code.Pals
        </div>
        <nav
          class="chakra-stack css-5ri2xm"
        >
          <a
            href="/"
            underline="none"
          >
            Home
          </a>
          <a
            href="/create"
            underline="none"
          >
            Create
          </a>
          <a
            href="/results"
            underline="none"
          >
            Search
          </a>
          <a
            href="/chatrooms"
            underline="none"
          >
            Chat
          </a>
          <a
            href="/aboutus"
            underline="none"
          >
            About Us
          </a>
          <a
            href="/login"
            underline="none"
          >
            Login
          </a>
          <div
            class="css-1lekzkb"
          >
            <div
              class="css-gmuwbf"
            >
              <div
                class="chakra-tabs css-13o7eu2"
              >
                <div
                  aria-orientation="horizontal"
                  class="chakra-tabs__tablist css-fgp5ep"
                  role="tablist"
                >
                  <button
                    aria-controls="tabs-1--tabpanel-0"
                    aria-selected="true"
                    class="chakra-tabs__tab css-2urs29"
                    data-index="0"
                    id="tabs-1--tab-0"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    One
                  </button>
                  <button
                    aria-controls="tabs-1--tabpanel-1"
                    aria-selected="false"
                    class="chakra-tabs__tab css-2urs29"
                    data-index="1"
                    id="tabs-1--tab-1"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    Two
                  </button>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </div>
      <div
        class="css-70qvj9"
      >
        <button
          class="chakra-button css-173s92d"
          type="button"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21.4,13.7C20.6,13.9,19.8,14,19,14c-5,0-9-4-9-9c0-0.8,0.1-1.6,0.3-2.4c0.1-0.3,0-0.7-0.3-1 c-0.3-0.3-0.6-0.4-1-0.3C4.3,2.7,1,7.1,1,12c0,6.1,4.9,11,11,11c4.9,0,9.3-3.3,10.6-8.1c0.1-0.3,0-0.7-0.3-1 C22.1,13.7,21.7,13.6,21.4,13.7z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          aria-controls="menu-list-3"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-button chakra-menu__menu-button css-hikdd6"
          id="menu-button-3"
          type="button"
        >
          <span
            class="css-xl71ch"
          />
        </button>
        <div
          class="css-r6z5ec"
          style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-1ozmk1d"
            id="menu-list-3"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <a
              href="/profile/undefined"
              underline="none"
            >
              's
              <hr
                aria-orientation="horizontal"
                class="chakra-menu__divider css-vfkdr8"
                role="separator"
              />
              Profile
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-0"
  >
    <div
      class="css-1mpt2dh"
    />
  </div>
</div>
`;
